### PR TITLE
docs: add the shell-autocompletion support in optimus cli

### PR DESCRIPTION
**Pull request by scortier**
### Description
1.  Automatically adds completion command for [shell completions](https://github.com/spf13/cobra/blob/master/shell_completions.md) for bash, zsh, fish and PowerShell. 
2. Users can configure the command auto creation:  
      1. disable the creation of the completion command
      2. disable completion descriptions
      3. disable the --no-descriptions flag for "always on" completion descriptions

### Preview
**Command to use : `./optimus [tab][tab]`**

Screenshot 2021-07-10 at 12 34 28 PM

### Related Issues 
Issue #32

### CODE REVIEW

**Comment by scortier** [comment]
@kushsharma  please review the PR

File: docs/guides/adding-shell-autocompletion.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+create   -- Create a new job/resource
+deploy   -- Deploy current project to server
+help     -- Help about any command
+render   -- convert raw representation of specification to consumables
+replay   -- re-running jobs in order to update data for older dates/partitions
+serve    -- Starts optimus service
+version  -- Print the client version information
+ ```
+
+## Additional Features :
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:82]: Remove all of this, why would user care about this?

File: docs/guides/adding-shell-autocompletion.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Shell Autocompletion Feature
+
+The shell auto-completion feature in optimus CLI provide the relief you not to remember the specific command , you just need to type some initials of the command then complete command gets generated automatically.If a completion command is already provided, uses that instead.
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:3]: People know what autocompletion is already, no need to explain

File: docs/guides/adding-shell-autocompletion.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## Steps to setup shell completion functionality 
+
+Before moving towards setup one can check whether the functionality is enabled in it's Optimus binary by executing `optimus [tab][tab]` in the CLI. If it's working fine then you are good to go.
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:7]: what to do mean here? what `functionality `

File: docs/guides/adding-shell-autocompletion.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+Otherwise, follow the setup shown below.
+
+### 1. Bash auto-completion 
+
+### Introduction :
+The optimus completion script for Bash can be generated with `./optimus completion bash`. Sourcing this script in your shell enables optimus completion.
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:13]: no need to specify `./` path of the optimus binary

**Review by kushsharma — CHANGES_REQUESTED** [review]

File: docs/guides/adding-shell-autocompletion.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+- Add the completion script to the /usr/local/etc/bash_completion.d directory:
+```
+# To load completions for each session, execute once:
+# Linux:
+$ ./optimus completion bash > /etc/bash_completion.d/_optimus
+# macOS:
+$ ./optimus completion bash > /usr/local/etc/bash_completion.d/_optimus
+```
+
+- If you installed optimus with Homebrew (as explained here), then the optimus completion script should already be in /usr/local/etc/bash_completion.d/_optimus. In that case, you don't need to do anything.
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:35]: link `as explained here`

File: docs/guides/adding-shell-autocompletion.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
+# Shell Autocompletion Feature
+
+The shell auto-completion feature in optimus CLI provide the relief you not to remember the specific command.If a completion command is already provided, uses that instead.
+
+## Steps to setup shell completion functionality 
+
+### 1. Bash auto-completion
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:7]: No need for numeric list

File: docs/guides/adding-shell-autocompletion.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+>Note: The Homebrew installation of bash-completion v2 sources all the files in the BASH_COMPLETION_COMPAT_DIR directory, that's why the latter two methods work.
+
+In any case, after reloading your shell, optimus completion should be working.
+
+
+### 2. Zsh Auto-completion
+
+The optimus completion script for Zsh can be generated with the command `optimus completion zsh`. Sourcing the completion script in your shell enables optimus autocompletion.
+
+- If shell completion is not already enabled in your environment,you will need to enable it.You can execute the following once:
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:46]: use proper text formatting like space after comma, space after `.`

File: docs/guides/adding-shell-autocompletion.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+If you get an error like complete:13: command not found: compdef, then add the following to the beginning of your ~/.zshrc file:
```
Comment (kushsharma) [inline docs/guides/adding-shell-autocompletion.md:47]: put error in back ticks

**Review by kushsharma — CHANGES_REQUESTED** [review]

File: docs/docs/reference/shell-autocompletion.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+- If shell completion is not already enabled in your environment, you will need to enable it. You can execute the following once:
+
+>If you get an error like `complete:13: command not found: compdef`, then add the following to the beginning of your `~/.zshrc` file:
+
+```
+  $ echo "autoload -U compinit; compinit" >> ~/.zshrc
+```
+- To load completions for each session, execute once:
+```
+  $ ./optimus completion zsh > "${fpath[1]}/_optimus"
```
Comment (kushsharma) [inline docs/docs/reference/shell-autocompletion.md:51]: remove `./` from all examples

**Pull request merged**